src/main/java/org/jboss/threads/EnhancedQueueExecutor.java

```diff
@@ -747,8 +752,20 @@ public Builder setMBeanName(final String mBeanName) {
     public void execute(Runnable runnable) {
         Assert.checkNotNullParam("runnable", runnable);
         final Runnable realRunnable = JBossExecutors.classLoaderPreservingTaskUnchecked(runnable);
-        int result;
-        result = tryExecute(realRunnable);
+        final int result;
+        try (TryExecuteTask tryExecuteTask = POOLED_COMBINER_TASKS.get().on(this, realRunnable)) {
```

Comment:
I wonder if something like #81 would allow us to avoid this threadlocal in favor of a couple additional references on an existing object?